__tests__/userController.tests.mjs

```diff
@@ -0,0 +1,93 @@
+import { signupUser, loginUser, logoffUser } from 'server/controllers/userController.mjs';
+import { mockRequest, mockResponse } from 'jest-mock-req-res';
+import admin from 'server/config/firebase.mjs';
+
+describe('User Controller', () => {
+    // Array to store all user IDs created during testing
+    const createdUserIds = [];
+    let testEmail;
+    const TEST_PASSWORD = 'password123';
+
+    // Helper function to create mock request and response objects
+    const createMockRequestResponse = (body) => {
```

Comment:
Total nitpick so feel free to ignore.

I think extracting this to a helper function makes the tests a little harder to read. It's not really saving us much if any typing, and it's not clear when I call the function from inside a test what the object I'm passing in is for. I would personally choose to explicitly call `mockRequst` and `mockResponse` in each test that wants them.

Reply:
I did try to explicitly call mockRequest and mockResponse, but I kept getting a “require is not defined” error. Even though we don’t have require in our code, the mockRequest and mockResponse package is trying to run our code in a CommonJS environment, which expects require to be available. Since we’re using ES modules and import instead, so I used helper function as a workaround.